app/controllers/posts_controller.rb

```diff
@@ -4,11 +4,43 @@ class PostsController < ApplicationController
   def index
     # Placeholder for all post page
     @user = User.find(params[:user_id]) # Assuming you're using user_id from params
-    @posts = @user.posts
+    @posts = @user.posts.includes(:comments).paginate(page: params[:page], per_page: 2)
   end
 
```

Comment:
- [ ] Let's use `CanCanCan for Post authorization in `Post Controller`
class PostsController < ApplicationController

    `load_and_authorize_resource`
    ...
  end